manticore/core/cpu/disasm.py

```diff
@@ -141,6 +138,18 @@ def unimplemented(self, il):
         return (il.operation == enums.LowLevelILOperation.LLIL_UNIMPL or
                 il.operation == enums.LowLevelILOperation.LLIL_UNIMPL_MEM)
 
+    def _get_current_llil_func(self, code, pc):
+        if pc in self.llil_func_cache:
```

Comment:
One thing you might want to consider is checking the target of an `LLIL_LOAD` operation, and removing any address from the cache that would be modified by that `LLIL_LOAD`. Otherwise, you could end up with cached instructions that do not match the underlying binary data.